### ISSUE TITLE
[wekan] guide is broken

### ISSUE THREAD

**Issue by luto** [Uberspace/lab#1095]
The guide for wekan does not work anymore. [explanation of what does not work here]. We added a banner to the guide to signal the broken state.

If you have any insights, please comment on this issue. Pull requests fixing the guide are also highly appreciated. If the guide does not get fixed within two months, we will remove it from the lab, as documented in the [broken guides policy](https://github.com/Uberspace/lab/blob/main/BROKEN_GUIDES.md).

```
wget wget https://releases.wekan.team/wekan-4.43.zip
--2021-08-22 22:32:23--  http://wget/
Resolving wget (wget)... failed: Name or service not known.
wget: unable to resolve host address ‘wget’
--2021-08-22 22:32:23--  https://releases.wekan.team/wekan-4.43.zip
Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving releases.wekan.team (releases.wekan.team)... 2606:4700:20::681a:859, 2606:4700:20::681a:959, 2606:4700:20::ac43:4537, ...
Connecting to releases.wekan.team (releases.wekan.team)|2606:4700:20::681a:859|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2021-08-22 22:32:23 ERROR 404: Not Found.
```

**Comment by luto** [comment]
@schoeke FYI

**Comment by schoeke** [comment]
Huh, seems like they changed the release channel. I try to update it within a week.

**Comment by unite42** [comment]
@luto @schoeke I followed the guide today with Wekan v5.72 and it worked.

**Comment by BabyFnord** [comment]
> Hello there, there is a [wekan installation+update script for Uberspace 7](https://github.com/BabyFnord/update_wekan) installations, alas it stopped working at wekan v5.28. I've lost contact with it's creator @kimdiallo, rendering the script stuck at an obsolete version of wekan (see [here for details](https://github.com/BabyFnord/update_wekan/issues/4)).
> 
> Quite an effort was put into this script, sadly fixing its code/professionally dealing with the prerequisites is beyond my skills. Not sure why its dev has stopped working on it, but I would appreciate and donate for getting it working again.
> 
> Anyone interested to pick this up?

FYI, I've fixed this plus a couple of related issues of said script. Now v1.8 performed a smooth update to wekan-v5.83. [Try it](https://github.com/BabyFnord/update_wekan/), if you like :)

**Comment by Thorgrimson** [comment]
I followed the guide with wekan v6.11 and got this error when I try start wekan:
```2022-03-07 11:05:55,649 DEBG 'wekan' stderr output:
Error: /lib64/libstdc++.so.6: version `CXXABI_1.3.9' not found (required by /home/thorgrim/bundle/programs/server/node_modules/fiber$
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1057:18)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Module.Mp.load (/home/thorgrim/bundle/programs/server/runtime.js:15:31)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/thorgrim/bundle/programs/server/node_modules/fibers/fibers.js:14:39)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Module.Mp._compile (/home/thorgrim/bundle/programs/server/runtime.js:50:23)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
```

**Comment by BabyFnord** [comment]
Which version of Node is active on your Uberspace? WeKan v6.11 requires Node v14.x, did you run `uberspace tools version use node 14` prior to installing v6.11?

**Comment by Thorgrimson** [comment]
It was 12. I start the install process new with 14. but there is still some errors:
this is the supervisord-log: https://haste.tchncs.de/ilozozukik.rb
I try to running the there mentionded fixed but got this error: https://haste.tchncs.de/duqamawazi.sql

**Comment by BabyFnord** [comment]
Try [this](https://github.com/BabyFnord/update_wekan/issues/4#issuecomment-980735972), maybe it will also help you.

**Comment by Thorgrimson** [comment]
works perfect ;)

**Comment by noave** [comment]
this guide is still marked as broken, I just tried to follow the guide using node14 but ended up in an error when starting the wekan service:

```
/home/isabell/bundle/programs/server/node_modules/fibers/future.js:280
						throw(ex);
						^

TypeError [ERR_INVALID_ARG_TYPE] [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at new NodeError (internal/errors.js:322:7)
    at validateString (internal/validators.js:124:11)
    at Object.join (path.js:1148:7)
    at module (models/attachments.js:13:22)
    at fileEvaluate (packages/modules-runtime.js:336:7)
    at Module.require (packages/modules-runtime.js:238:14)
    at Module.moduleLink [as link] (/home/isabell/bundle/programs/server/npm/node_modules/meteor/modules/node_modules/reify/lib/runtime/index.js:52:22)
    at module (server/publications/attachments.js:1:24)
    at fileEvaluate (packages/modules-runtime.js:336:7)
    at Module.require (packages/modules-runtime.js:238:14)
    at require (packages/modules-runtime.js:258:21)
    at /home/isabell/bundle/programs/server/app/app.js:32035:1
    at /home/isabell/bundle/programs/server/boot.js:401:38
    at Array.forEach (<anonymous>)
    at /home/isabell/bundle/programs/server/boot.js:226:21
    at /home/isabell/bundle/programs/server/boot.js:464:7
    at Function.run (/home/isabell/bundle/programs/server/profile.js:280:14)
    at /home/isabell/bundle/programs/server/boot.js:463:13 {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

do you have a fix for this? Otherwise we need to remove the guide from the lab soon

**Comment by BabyFnord** [comment]
@noave Oh, another `node fibers` issue 🙃. Maybe you want to try [this](https://github.com/BabyFnord/update_wekan/issues/4#issuecomment-980735972) and report back, if this helps?

**Comment by noave** [comment]
unfortunately this didn't fix it for me

```
rm node_modules/.bin/node-gyp
rm node_modules/.bin/node-pre-gyp
npm uninstall fibers
npm install fibers
npm audit fix --force
```

do you think you could try install wekan following the guide on a fresh account @BabyFnord and fix these issues to make the guide working again? It looks like you know the app quite well :slightly_smiling_face:

**Comment by BabyFnord** [comment]
I am less savvy than you think, plus I am not an Uberspace team member. Having said that, I do not have access to other Spaces other than my own. As always, log file output or more detailed feedback beyond 
> this didn't fix it for me
increases chances of getting help (or getting anywhere, for that matter 😉).

**Comment by noave** [comment]
> As always, log file output or more detailed feedback beyond .. increases chances of getting help (or getting anywhere, for that matter).

sorry, I did not mention that the same error like posted here https://github.com/Uberspace/lab/issues/1095#issuecomment-1116272216 stays even after rebuilding fibers.

But this very much sounds like we need to remove the guide unless someone can fix these dependency and service issues. I have set a reminder for next month

**Comment by BabyFnord** [comment]
Got it, @noave. Yet, as I already mentioned, as of v6.11 (I was wrong with 5.18), WeKan can not be properly started as a service via `supervisord` _anymore_ and your mileage varies wildly staying with a service-run installation. See here for my [journey](https://github.com/wekan/wekan/issues/4393#issuecomment-1086896192). Hence I switched to opening a `tmux` session in order to run `start-wekan.sh`. Kinda repeating myself here but, WeKan expects parameters to be passed as in `start-wekan.sh`.  The .ini as per this manual is deprecated for the time being, until someone works out a compatible revision.

**Comment by friesenkiwi** [comment]
@BabyFnord do I understand correctly from this thread here and https://github.com/wekan/wekan/issues/4393#issuecomment-1090067565, that currently the now-removed guide at https://raw.githubusercontent.com/Uberspace/lab/530c74697dc67ca95e4627fc54369ee19839b433/source/guide_wekan.rst generally works fine with the latest Wekan on latest Uberspace 7, just the actual starting/running via `supervisord` doesn't work, but using `tmux` all is fine? And all is also wrapped in https://github.com/BabyFnord/update_wekan/blob/main/update_wekan2 ?

**Comment by BabyFnord** [comment]
Well: yes, no, maybe, @friesenkiwi 😅. Sorry for a late reply, tho. 

First off, I'm a prosumer with a very limited understanding of any programming language and Linux environment. 
Second, the update script has been coded by an Uberspace team member, my contributions to this are, well, limited too. 
Third, my WeKan ƒ structure has grown to differ to the gone UberLab. 
Fourth (you may find this more interesting than the aforementioned 1-3), the previous method of parsing the current WeKan version stopped working for reasons beyond me. I fail at fixing the issue. The script version I use successfully at this time, needs the current version number _manually_ inserted, and, subsequently, WeKan can only be started per `tmux` and a slightly modified start-script. The caveat is that, if WeKan would ever crash (which it only did once in my use case, long ago), not using the old `supervisord` way means, you have to manually restart it by `tmux`'ing the start-script.

Long story short: 
It works for me. 
Chances are, I might fail to make it work for someone else, but I'm willing to share what I have and know.
At best, someone _skilled_ would & could update the `update_script`. Accordingly a minimal UberLab WeKan manual would make it fairly easy for the rest of us, to install and update WeKan, which is a way kewl and super-useful FOSS Kanban board btw.!
